File: BaekJoon/Queue/큐2_18258.swift

// MARK: TLE답안

/*
struct DoubleStackQueue<T> {
    var leftStack = [T]()
    var rightStack = [T]()
    
    mutating func push(_ val: T) {
        rightStack.append(val)
    }
    
    mutating func pop() {
        if leftStack.isEmpty && rightStack.isEmpty { print(-1); return }
        if leftStack.isEmpty {
            leftStack = rightStack.reversed()
            rightStack = []
        }
        print(leftStack.last!)
        leftStack.removeLast()
    }
    
    func size() {
        print(leftStack.count + rightStack.count)
    }
    
    func front() {
        if leftStack.isEmpty && rightStack.isEmpty { print(-1); return }
        if leftStack.isEmpty {
            print(rightStack.first!)
        } else {
            print(leftStack.last!)
        }
    }
    
    func back() {
        if leftStack.isEmpty && rightStack.isEmpty { print(-1); return }
        if leftStack.isEmpty {
            print(rightStack.last!)
        } else {
            print(leftStack.first!)
        }
    }
    
    func isEmpty() {
        print(leftStack.isEmpty && rightStack.isEmpty ? 1 : 0)
    }
}

let T = Int(readLine()!)!
var DQ = DoubleStackQueue<String>()
for _ in 0..<T {
    let arr = readLine()!.split(separator: " ").map { String($0) }
    if arr[0] == "push" { DQ.push(arr[1]) }
    else if arr[0] == "front" { DQ.front() }
    else if arr[0] == "back" { DQ.back() }
    else if arr[0] == "size" { DQ.size() }
    else if arr[0] == "empty" { DQ.isEmpty() }
    else if arr[0] == "pop" { DQ.pop() }
}
*/

File: BaekJoonPS/BaekjoonAlgorithmPS/Queue/큐2_18258.cpp

// MARK: - 큐2 18258

// MARK: Queue C++ 문제풀이

#include <iostream>
#include <queue>
#include <string>
using namespace std;

queue<int> Q;

void runQ() {
  string key;
  int value;
  cin>>key;
  if(key == "push") {
    cin>>value;
    Q.push(value);
  } else if(key == "pop") {
    if(Q.empty()) printf("-1\n");
    else {
      printf("%d\n",Q.front());
      Q.pop();
    }
  } else if(key == "size") {
    printf("%ld\n",Q.size());
  } else if(key == "empty") {
    printf("%d\n",Q.empty() ? 1 : 0);
  } else if(key == "front") {
    printf("%d\n",Q.empty() ? -1 : Q.front());
  } else if(key == "back") {
    printf("%d\n",Q.empty() ? - 1 : Q.back());
  }
}

int main() {
  ios_base :: sync_with_stdio(0); cin.tie(0);
  int N; cin>>N;
  for (int i=0; i<N; i++) {
    runQ();
  }
}
